### PR TITLE
PR: move and combine upgrade docs

### LINKED ISSUE #17: [DOC] How to guide for upgrading Nebari
Should be written in tandem with https://github.com/nebari-dev/nebari/issues/16

[notion doc](https://www.notion.so/quansightlabs/Upgrade-Nebari-09b61a73c5274c60b18f4eeb6e5c95bd)

### PR DESCRIPTION
**Pull request by ppwadhwa**
This PR attempts to move the qhub upgrade docs over to nebari.  It also combines the doc for regular upgrading and the doc for breaking changes upgrading.  I am not sure if this is still an issue, but this is my attempt to include both in one.

Fixes #17 

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

**Comment by kcpevey** [comment]
@ppwadhwa can you check the pre-commit failure?

**Review by iameskild — COMMENTED** [review]
Hi @ppwadhwa, thanks for helping update the docs! Everything looks factually correct and is well written. 

My only concern is how relevant these particular docs are going forward. The `upgrade` command is something that we added for the pre-`0.4.0` to `0.4.0` upgrade given all of the changes to the config file. Going forward, do we want to continue to support this type of command? This is likely a question that requires a bit more discussion and I apologize for not recognizing that sooner. 

@kcpevey @trallard @costrouc 
What are your thoughts about retaining this `upgrade` command going forward? If we don't continue with this pattern, is it still worth adding docs like these?

**Comment by kcpevey** [comment]
I think we are still close enough to the 0.3.0 series that we need to continue to maintain these docs. It'll be easy to drop them in the future if the team decides to deprecate the `upgrade` function. Since deprecating `upgrade` would be an RFD, I'd say let's move forward with this PR.

**Comment by iameskild** [comment]
Hey @ppwadhwa and @kcpevey, someone recently asked me if we had docs on how to perform a basic Nebari upgrade (say from `2022.11.1` to `2023.1.1`) and I realize we don't have this in our old or new docs. I think that would be a great addition to this PR if you're up for it. 

Essentially, I see two main headers for this `# Upgrading Nebari` doc:
1. Regular Upgrade (regular maintenance, no breaking changes)
  - simple instructions to upgrade from one CalVer version to the next would include
    - `pip` or `conda` install latest Nebari version
    - update `nebari_version: xxx` in the nebari-config.yaml to this latest version
    - (run `nebari validate -c nebari-config.yaml`) optional but best-practice
    - run `nebari render -c nebari-config.yaml` to rerender all the files
    - finally ru `nebari deploy -c nebari-config.yaml` 
      - if deploying via GitOps, you would just open a PR on the deployment repo and merge once the linter passes.
2. Upgrade Nebari from pre-0.4.0 to 0.4.0 
  - these would be the docs you included above, no changes needed.

File: docs/docs/how-tos/nebari-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
+---
+id: nebari-upgrade
+title: Upgrade Nebari
+description: An overview of how to upgrade Nebari. Includes instructions on updating when breaking changes are expected to occur.
+---
+
+# Upgrading Nebari
```
Comment (kcpevey) [inline docs/docs/how-tos/nebari-upgrade.md:7]:
```suggestion
# Upgrade Nebari from pre-0.4.0 to 0.4.0
```

**Comment by kcpevey** [comment]
@iameskild I think that's a great suggestion. I've suggested the change to make this PR be the addition of upgrading from pre-0.4.0 to 0.4.0. Then I'd say lets open a separate issue and pr for the "Upgrading minor versions" (to be included in this same document.

File: docs/docs/how-tos/nebari-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+This is a guide to upgrade Nebari to a newer version.
+
+:::note Breaking upgrades
```
Comment (kcpevey) [inline docs/docs/how-tos/nebari-upgrade.md:11]: I think all these notes should be warnings instead.

**Review by kcpevey — APPROVED** [review]

**Pull request merged**